core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java

```diff
@@ -397,10 +408,10 @@ public DetailedEnvironmentResponse registerExternalDatalake(String environmentNa
                     Map<String, Map<String, String>> serviceSecretParamMap = StringUtils.isEmpty(registerDatalakeRequest.getRangerAdminPassword())
                             ? new HashMap<>() : Map.ofEntries(Map.entry(ServiceDescriptorDefinitionProvider.RANGER_SERVICE, Map.ofEntries(
                             Map.entry(ServiceDescriptorDefinitionProvider.RANGER_ADMIN_PWD_KEY, registerDatalakeRequest.getRangerAdminPassword()))));
-                    DatalakeResources datalakeResources = datalakeConfigProvider.collectAndStoreDatalakeResources(environmentName, datalakeAmbariUrl,
+                    DatalakeResources datalakeResources = datalakeConfigProvider.collectAndStoreDatalakeResources(environmentName, envView, datalakeAmbariUrl,
                             ambariUrl.getHost(), ambariUrl.getHost(), ambariClient, serviceSecretParamMap, ldapConfig, kerberosConfig, rdssConfigs,
                             environment.getWorkspace());
-                    environment.setDatalakeResources(datalakeResources);
+                    environment.setDatalakeResources(Sets.newHashSet(datalakeResources));
```

Comment:
shouldn't this be like: environment.getDatalakeResources().add(datalakeResources) ?